### ISSUE TITLE
Example from readme does not work

### ISSUE THREAD

**Issue by everflux** [elastic/elasticsearch-groovy#31]
I am using a maven groovy project with elasticsearch-groovy 2.0.0. The example from the readme does not work due to multiple problems.
The original example
```
TransportClient client = new TransportClient(Settings.settingsBuilder {
  client.transport.sniff = true
  cluster.name = "your-cluster-name"
})
```
needed to be changed to

```
        client = new TransportClient(Settings.settingsBuilder{
                client {
                    transport {
                        sniff = true
                    }
                }
                cluster {
                    name = "your-cluster-name"
                }
            })
```
otherwise the error ``Exception in thread "main" java.lang.NullPointerException: Cannot get property 'transport' on null object`` was thrown.
After that the following exception is thrown ``Exception in thread "main" groovy.lang.GroovyRuntimeException: Could not find matching constructor for: org.elasticsearch.client.transport.TransportClient(org.elasticsearch.common.settings.Settings$Builder)``

When checking the TransportClient class, there is indeed no constructor accepting a builder, there is no constructor at all. As I am not very familiar with groovy yet, this might be how the magic should work, accessing the builder to create a new TransportClient instance behind the scenes, but it fails and I have no clue what I need to change to get it working.

**Comment by everflux** [comment]
This is what worked for me finally

```
        def settings = Settings.settingsBuilder
        {
            cluster {
                name = clusterName
            }
            client {
                transport {
                    sniff = true
                }
            }
        };
        client = TransportClient.builder().settings(settings).build()
        client.addTransportAddress(new InetSocketTransportAddress(InetAddress.getByName("localhost"), 9300))
```

**Comment by pickypg** [comment]
Thanks for pointing this issue out. In ES 2.0 (and later), the `TransportClient` changed to require the builder pattern to use it. I've updated the README to show its usage.

You should be able to use the original version, but change it from:

```groovy
TransportClient client = new TransportClient(Settings.settingsBuilder {
  client.transport.sniff = true
  cluster.name = "your-cluster-name"
})
```

to:

```groovy
TransportClient client = TransportClient.builder().settings(Settings.settingsBuilder {
  client.transport.sniff = true
  cluster.name = "your-cluster-name"
}).build()
```

The `Closure` should not be impacted at all.

**Comment by pickypg** [comment]
Closed by 7f86551cf560be3f6d903ecba5d5fbe0ae780468
